### PR TITLE
Allow any object as an argument for random.seed

### LINKED ISSUE #3411: random.seed does not accept floats
On CPython/PyPy, this works:

```python
>>>>> import random
>>>>> random.seed(0.5)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/.../RustPython/vm/pylib-crate/Lib/random.py", line 133, in seed
    super().seed(a)
TypeError: 'float' object cannot be interpreted as an integer
```

### PR DESCRIPTION
**Pull request by DimitrisJim**
Closes #3411

### CODE REVIEW

File: Lib/test/test_random.py [hunk 1]
```diff
@@ -899,7 +883,7 @@ def test_gammavariate_alpha_greater_one(self, random_mock):
         returned_value = random.gammavariate(1.1, 2.3)
         self.assertAlmostEqual(returned_value, 2.53)
 
-    # TODO: RUSTPYTHON
+    # TODO: RUSTPYTHON
```
Comment (fanninpm) [inline Lib/test/test_random.py:886]: Was this just a failure?

**Review by fanninpm — COMMENTED** [review]
Sometimes you added only spaces to the ends of comments.

File: Lib/test/test_random.py (hunk 1 above)
Comment (DimitrisJim) [reply Lib/test/test_random.py:886]: I think both these errors had to do with precision issues (an assert failed) so I was slightly hesitant to add the output.

Comment (DimitrisJim) [reply Lib/test/test_random.py]: went ahead and just added that the assert fails in these cases.

**Review by youknowone — APPROVED** [review]

**Pull request merged**